src/liquidity.rs

```diff
@@ -198,54 +198,86 @@ where
 	pub(crate) async fn lsps2_receive_to_jit_channel(
 		&self, amount_msat: Option<u64>, description: &str, expiry_secs: u32,
 		max_total_lsp_fee_limit_msat: Option<u64>,
-	) -> Result<(Bolt11Invoice, u64), Error> {
+		max_proportional_lsp_fee_limit_ppm_msat: Option<u64>,
+	) -> Result<(Bolt11Invoice, Option<u64>, Option<u64>), Error> {
 		let lsps2_service = self.lsps2_service.as_ref().ok_or(Error::LiquiditySourceUnavailable)?;
 
 		let fee_response = self.request_opening_fee_params().await?;
 
-		if let Some(amount_msat) = amount_msat {
+		let (min_total_fee_msat, min_prop_fee_ppm_msat, min_opening_params) = if let Some(
+			amount_msat,
+		) = amount_msat
+		{
+			// `MPP+fixed-invoice` mode
```

Comment:
Would it be possible to break this into two functions (and possibly refactored further), calling the appropriate one as necessary? Likewise for `receive_payment_via_jit_channel_inner` as would be necessary. Otherwise, you have two `Option` parameters passed in where only one can be `Some`. Similarly with the variables being assigned here.

Generally, it's more difficult to follow having one function to handle all the possibilities but has assumptions about the `Option` parameters and therefore which code paths are hit.

Reply:
Thanks, that LGTM.